Source/WPEFramework/ControllerJsonRpc.cpp

```diff
@@ -547,6 +559,198 @@ namespace Plugin {
         return result;
     }
 
+    // Property: version of the controller
+    // Return codes:
+    //  - ERROR_NONE: Success
+    //  - ERROR_GENERAL: Failed to get the version
+    uint32_t Controller::get_version(Core::JSON::String& response) const
+    {
+        uint32_t result = Core::ERROR_NONE;
+   
+        ASSERT(_pluginServer != nullptr);
+        
+        if (_pluginServer !=nullptr)  {
+            response = _pluginServer->_config.Version();
```

Comment:
Sorry for not noticing earlier. It is not the idea to use private members from another class, in this case _config, it breaks the idea of encapsulation. The reason it even compiles here is that there is that controller is a fiend class, guess to allow it to access a private method that should not be accessible by something else then the Controller class (indeed I guess a more specific friend to only allow the method inside controller using it access would have been better bit that is sometimes harder to achieve). But there is a public Configuration method to access the configuration, better to use that instead of _config (e.g. when somebody would rename _config, implement it differently, e.g. add lazy loading, make a aggregate via a pointer then unexpectedly the code would break because of the above usage of the private _config data member). And of course this goes also for the usage in the other methods below.